dataprofiler/labelers/data_processing.py

```diff
@@ -1434,7 +1438,8 @@ def process(self, data, results, label_mapping):
         pad_label = self._parameters['pad_label']
 
         # Format predictions
-        results = self.match_sentence_lengths(data, copy.deepcopy(results),
+        # FORMER DEEPCOPY, SHALLOW AS ONLY INTERNAL
+        results = self.match_sentence_lengths(data, dict(results),
```

Comment:
i'm concerned that results is a nested dict and this will alter the input.

To me, if this is the functionality we wanted, it would have been better to process without a return so the user knows that it is altering the value.

Not sure if a user ever uses the processor directly, but in this case they would unknowingly end up with an altered input

Reply:
Personally, I think if a user is creating their own data processing pipeline they can deepcopy in their pipeline. They would likely find the issue if they were doing it. That being said, that's not a use case I think we need to prioritize.